src/python/py/models/builder.py

```diff
@@ -924,7 +934,16 @@ def make_rotary_embedding_caches(self, rotemb, **kwargs):
         if self.rotemb_attrs["create_rotary_embedding_caches"]:
             if not hasattr(rotemb, "cos_cached"):
                 # Create cos/sin caches if not already created
-                cos_cache, sin_cache = self.make_rotary_embedding_caches_from_scratch()
+                if self.ep == "dml":
+                    cos_cache_large, sin_cache_large = self.make_rotary_embedding_caches_from_scratch()
+                    self.rotemb_attrs["rescale_factors"] = self.rotemb_attrs["multi_cache"]["short_factor"]
+                    self.rotemb_attrs["cache_length"] = self.original_context_length
+                    self.rotemb_attrs["mscale"] = self.rotemb_attrs["multi_cache"]["short_mscale"]
```

Comment:
These changes will prevent non-Phi-3.5 models from being created for the DML EP. Can we move these changes inside the `Phi3Mini128KModel` class, override the `make_rotary_embedding_caches` method, and guard them with the `if self.ep == "dml"` check?